internal/agent/status/status.go

```diff
@@ -154,23 +177,52 @@ func (r *nrReporter) ReportEntity() (re ReportEntity, err error) {
 	}, nil
 }
 
+func (r *nrReporter) ReportHealth() HealthReport {
+	agentID := r.idProvide().ID.String()
+
+	return r.getHealth(agentID)
+}
+
+func (r *nrReporter) getHealth(agentID string) HealthReport {
```

Comment:
I would add a comment about the goal of this ~:
`Make an http req to the command api to validate the ingest key is valid and connectivity is ok`

Reply:
Added in last commit